index.js

```diff
@@ -17,12 +17,18 @@ if (testSupport()) {
   storageEngine = localStorage
 }
 
+let pageShowHandler
+
 export let windowPersistentEvents = {
-  addEventListener(key, listener) {
+  addEventListener(key, listener, sync) {
+    pageShowHandler = () => sync()
+
     window.addEventListener('storage', listener)
+    window.addEventListener('pageshow', pageShowHandler)
   },
   removeEventListener(key, listener) {
```

Comment:
Let’s add `sync` here and we will be able to remove `pageShowHandler`.

It will reduce project’s size and reduce hidden state.

Reply:
Done